app/components/bottom-bar.tsx

```diff
@@ -1,19 +1,28 @@
-export type BottomBarProps = Omit<JSX.IntrinsicElements["div"], "style">
-export default function BottomBar({ children, className = "", ...rest }: BottomBarProps) {
+export type BottomBarProps = { isSliderOpen?: Boolean } & Omit<
```

Comment:
I think this ties the concept of shrinking the container to the fact that the slider is open or closed, and that we require the parent component to know this as well.  What happens if other components need to react to the sidebar open state?

I think we can do better here. We can wrap the app in a context that manages the sidebar concept, and manage the width internally.

```tsx
// Inside this component
const [state, toggle] = useContext(SidebarContext);
```

React accordingly depending on the value of `state`.

Reply:
Got it, it'll be updated within the next commit!